components/brave_wallet_ui/common/hooks/timeout.ts

```diff
@@ -10,32 +10,58 @@ export default function useTimeout (
   isWalletLocked: boolean
 ) {
   const [lastCheckedTime, setLastCheckedTime] = React.useState<number>()
-  const [currentMousePosition, setCurrentMousePosition] = React.useState({ x: 0, y: 0 })
-  const [lastMousePosition, setLastMousePosition] = React.useState({ x: 0, y: 0 })
+  const hasInteractedSinceInterval = React.useRef<boolean>(false)
+
   React.useMemo(() => {
-    const { x, y } = currentMousePosition
-    // Checks to see if the mouse moved everytime lastCheckTime is updated
-    if (lastMousePosition.x !== x || lastMousePosition.y !== y) {
-      setLastMousePosition({ x, y })
-      // If mouse has moved will notify keyring here.
+    // Checks to see if the hasInteractedSinceInterval is
+    // true or false everytime lastCheckTime is updated
+    if (hasInteractedSinceInterval.current) {
+      hasInteractedSinceInterval.current = false
+      // If true will notify keyring here.
       notifyUserInteraction()
     }
   }, [lastCheckedTime])
 
+  const onMouseMove = () => {
+    if (!hasInteractedSinceInterval.current) {
+      hasInteractedSinceInterval.current = true
+    }
+  }
+
+  const onKeyDown = () => {
+    if (!hasInteractedSinceInterval.current) {
+      hasInteractedSinceInterval.current = true
+    }
+  }
+
+  const onVisibilityChange = () => {
+    if (document.visibilityState === 'visible' && !hasInteractedSinceInterval.current) {
+      hasInteractedSinceInterval.current = true
+    }
+  }
+
   React.useEffect(() => {
-    // Starts tracking mouse move events when wallet is unlocked
+    const removeEventListeners = () => {
+      window.removeEventListener('mousemove', onMouseMove)
+      window.removeEventListener('keydown', onKeyDown)
+      document.removeEventListener('visibilitychange', onVisibilityChange)
+    }
+
+    // Starts tracking mousemove, keydown and visibilitychange events when wallet is unlocked
     if (!isWalletLocked) {
-      const mousePosition = (event: MouseEvent) => setCurrentMousePosition({ x: event.clientX, y: event.clientY })
-      window.addEventListener('mousemove', mousePosition)
-      return () => window.removeEventListener('mousemove', mousePosition)
+      window.addEventListener('mousemove', onMouseMove)
+      window.addEventListener('keydown', onKeyDown)
+      document.addEventListener('visibilitychange', onVisibilityChange)
+    } else {
+      removeEventListeners()
     }
-    return
-  }, [isWalletLocked])
+    return removeEventListeners
+  }, [isWalletLocked, hasInteractedSinceInterval])
 
   React.useEffect(() => {
     // If wallet is unlocked, will start interval
     // and setLastCheckedTime every 50 seconds to trigger
-    // a check to see if the mouse moved.
+    // a check to see if hasInteractedSinceInterval is true.
     if (!isWalletLocked) {
       const interval = setInterval(() => {
         const date = Date.now()
```

Comment:
Don't think we need to bother with state and then wait for another render round, simply do the check here, no?
```ts
    if (hasInteractedSinceInterval.current) {
      hasInteractedSinceInterval.current = false
      notifyUserInteraction()
    }
```

Reply:
Updated!